components/dashboard/src/Setup.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useState } from "react";
+import Modal from "./components/Modal";
+import { getGitpodService, gitpodHostUrl } from "./service/service";
+import { GitIntegrationModal } from "./settings/Integrations";
+
+export default function Setup() {
+
+    const [showModal, setShowModal] = useState<boolean>(false);
+
+    useEffect(() => {
+        (async () => {
+            const dynamicAuthProviders = await getGitpodService().server.getOwnAuthProviders();
+            const previous = dynamicAuthProviders.filter(ap => ap.ownerId === "no-user")[0];
+            if (previous) {
+                await getGitpodService().server.deleteOwnAuthProvider({ id: previous.id });
+            }
+        })();
+    }, []);
+
+    const acceptAndContinue = () => {
+        setShowModal(true);
+    }
+
+    const onAuthorize = (payload?: string) => {
+        // run without await, so the integrated closing of new tab isn't blocked
+        (async () => {
+            window.location.href = gitpodHostUrl.asDashboard().toString();
+        })();
+    }
+
+    const headerText = "Configure a git integration with a GitLab or GitHub."
```

Comment:
**nitpick**: Minor rephrasing.

```suggestion
    const headerText = "Configure a git integration with a GitLab or GitHub instance."
```

Alternatively, this we could just drop the `a`. You call! 📞 

```suggestion
    const headerText = "Configure a git integration with GitLab or GitHub."
```